File: 551-student-attendance-record-i/student-attendance-record-i.go
func checkRecord(s string) bool {
    absent, late := 0, 0
	for i := range s {
		if s[i] == 'A' {
			absent++

			if absent == 2 {
				return false
			}
		}

		if s[i] == 'L' {
			late++

			if late == 3 {
				return false
			}
		} else {
			late = 0
		}
	}
	return true
}

File: 551-student-attendance-record-i/student-attendance-record-i.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        absentCount = 0
        lateCount = 0
        for i in range(len(s)):
            if s[i] == "A":
                absentCount += 1
                if absentCount == 2:
                    return False

            if s[i] == "L":
                lateCount += 1
                if lateCount == 3:
                    return False
            else:
                lateCount = 0

        return True